nameko/rpc.py

```diff
@@ -194,12 +195,16 @@ def send_response(self, result, exc_info, **kwargs):
         if exc_info is not None:
             error = serialize(exc_info[1])
 
-        # disaster avoidance serialization check: `result` must be json
+        # disaster avoidance serialization check: `result` must be
         # serializable, otherwise the container will commit suicide assuming
         # unrecoverable errors (and the message will be requeued for another
         # victim)
+
+        serializer = self.config.get(
+            SERIALIZER_CONFIG_KEY, DEFAULT_SERIALIZER)
+
         try:
-            json.dumps(result)
+            kombu.serialization.dumps(result, serializer)
         except Exception:
```

Comment:
We need to catch everything here, to make sure we avoid the nasty suicide and requeue problem.

If ``serializer`` is a bogus value for example, kombu will probably throw something other than nice serialization error, and we want to catch it regardless.